File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    
    let vowels = ['a','e','i','o','u','A', 'E', 'I', 'O', 'U']
    let mapa = new Map()
    let mapb = new Map()
    let i = 0, j = s.length-1
    while(i < j) {
        let cha = s[i], chb = s[j]
        mapa.set(cha, (mapa.get(cha) || 0 ) + 1)
        mapb.set(chb, (mapb.get(chb) || 0 ) + 1)
        i++
        j--
    }
    let count1 = 0, count2 = 0
    for(let v of vowels) {
        count1 = count1 + (mapa.get(v) || 0)
        count2 = count2 + (mapb.get(v) || 0)
    }
    return count1 == count2
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        Set<Character> set = new HashSet<>();
        set.add('a');
        set.add('e');
        set.add('i');
        set.add('o');
        set.add('u');
        s = s.toLowerCase();
        int count1 = 0, count2 = 0;
        for(int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if(set.contains(ch)) {
                if(i < s.length() / 2) {
                   count1++;
                } else {
                    count2++;
                }
            }
        }
        return count1 == count2;
    }
}